src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -22,84 +22,64 @@ sealed class BlobFeedAction
         private static readonly CancellationToken CancellationToken = TokenSource.Token;
 
         public BlobFeed feed;
-        public int MaxClients { get; set; } = 8;
         
         const string feedRegex = @"(?<feedurl>https:\/\/(?<accountname>[^\.-]+)(?<domain>[^\/]*)\/((?<token>[a-zA-Z0-9+\/]*?\/\d{4}-\d{2}-\d{2})\/)?(?<containername>[^\/]+)\/(?<relativepath>.*)\/)index\.json";
 
-        public BlobFeedAction(string expectedFeedUrl, string accountKey, string indexDirectory, MSBuild.TaskLoggingHelper Log)
+        public BlobFeedAction(string accountName, string accountKey, string containerName, string packagesPath, MSBuild.TaskLoggingHelper Log)
         {
             this.Log = Log;
-            Match m = Regex.Match(expectedFeedUrl, feedRegex);
-
-            if (m.Success)
-            {
-                string accountName = m.Groups["accountname"].Value;
-                string containerName = m.Groups["containername"].Value;
-                string relativePath = m.Groups["relativepath"].Value;
-                string feedUrl = m.Groups["feedurl"].Value;
-
-                bool isPublic = string.IsNullOrWhiteSpace(m.Groups["token"].Value);
-                this.feed = new BlobFeed(accountName, accountKey, containerName, relativePath, feedUrl, string.IsNullOrWhiteSpace(indexDirectory) ? Path.GetTempPath() : indexDirectory, Log, isPublic);
-            }
-            else
-            { 
-                throw new Exception("Unable to parse expected feed. Please check ExpectedFeedUrl.");
-            }
+            this.feed = new BlobFeed(accountName, accountKey, containerName, packagesPath, Log);
```

Comment:
This now means we are going to need to duplication information in our build definitions. One URL for restore and then a bunch of container names and such that needs to match for the publishing. I would much prefer if we can have just one set of information that doesn't need to be duplicated and remain in sync.